libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -84,20 +80,17 @@ public static void AddBotRuntime(this IServiceCollection services, IConfiguratio
                     o.RootDialog = rootDialog;
                 });
 
-            // ResourceExplorer. TryAddSingleton will only add if there is no other registration for resource explorer.
-            // Tests use this to inject custom resource explorers but could also be used for advanced runtime customization scenarios.
-            services.TryAddSingleton<ResourceExplorer>(serviceProvider =>
-                new ResourceExplorer()
-                    .AddFolder(applicationRoot)
-                    .RegisterType<OnQnAMatch>(OnQnAMatch.Kind));
+            // Resource explorer
+            services.AddSingleton<ResourceExplorerOptions>();
+            services.TryAddSingleton<ResourceExplorer>(sp => new ResourceExplorer(sp.GetRequiredService<ResourceExplorerOptions>()).AddFolder(applicationRoot));
```

Comment:
Move to `ConfigurationResourceExplorer`, less DI in resource explorer itself

Reply:
Just submitted a change to use `ConfigurationREsourceProvider` which simplified this part and reduced the DI usage